fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/view/LaoView.java

```diff
@@ -0,0 +1,93 @@
+package com.github.dedis.popstellar.model.objects.view;
+
+import com.github.dedis.popstellar.model.objects.*;
+import com.github.dedis.popstellar.model.objects.security.MessageID;
+import com.github.dedis.popstellar.model.objects.security.PublicKey;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public final class LaoView {
+
+  private final Lao lao;
+
+  /**
+   * This class offers useful getters for Lao state to handlers and prevents changing its state. It
+   * is provided as an intermediate step towards functional handling of Objects. To change the state
+   * of an Lao, one can use getLao() which returns a copy of the wrapped Lao, and update the
+   * repository with said updated LAO.
+   *
+   * @param lao the Lao to be wrapped
+   */
+  public LaoView(Lao lao) {
+    if (lao == null) {
+      throw new IllegalArgumentException();
+    }
+    this.lao = new Lao(lao);
+  }
+
+  public Lao getLao() {
+    return new Lao(lao);
+  }
+
+  public long getLastModified() {
+    return lao.getLastModified();
+  }
+
+  public String getName() {
+    return lao.getName();
+  }
+
+  public String getId() {
+    return lao.getId();
+  }
+
+  public boolean areWitnessSetsEqual(Set<PublicKey> witnesses) {
+    return lao.getWitnesses().equals(witnesses);
+  }
+
+  public boolean isWitnessesEmpty() {
+    return lao.getWitnesses().isEmpty();
+  }
+
+  public boolean isOrganizer(PublicKey publicKey) {
+    return lao.getOrganizer().equals(publicKey);
+  }
+
+  public boolean isWitness(PublicKey publicKey) {
+    return lao.getWitnesses().contains(publicKey);
+  }
+
+  public Channel getChannel() {
+    return new Channel(lao.getChannel());
+  }
+
+  public Optional<RollCall> getRollCall(String id) {
+    Optional<RollCall> optional = lao.getRollCall(id);
+    return optional.map(RollCall::new); // If empty returns empty optional, if not
+    // returns optional with copy of retrieved RollCall
+  }
+
+  public Optional<Chirp> getChirp(MessageID messageID) {
+    Optional<Chirp> optional = lao.getChirp(messageID);
+    return optional.map(Chirp::new); // If empty returns empty optional, if not
+    // returns optional with copy of retrieved Chirp
+  }
```

Comment:
is the idea that later on, `getChirp` and other feature-specific functions will be somewhere other than the `LaoView` ?

Reply:
We envision down the road to remove LaoView as it would not be needed anymore. We would like to end up with a functional handling of the objects. Furthermore we would like to break down the Lao object, it's too big and does too many things.

Reply:
makes sense !